### ISSUE TITLE
Unable to build apk on CI after updating to 1.22.5

### ISSUE THREAD

**Issue by Sadoge** [flutter/flutter#73467]
When I am trying to build an APK (debug) on a linux docker image hosted on GitLab, I get two errors. I already fixed an issue with the two plugins but they both say this is not an issue with their plugin but a flutter issue. So that is why I am making it here.

<details><summary>Debug logs</summary>

```console
[  +90 ms] > Task :background_fetch:generateDebugBuildConfig
[        ] > Task :background_fetch:javaPreCompileDebug
[+3099 ms] > Task :background_fetch:compileDebugJavaWithJavac
[        ] > Task :background_fetch:bundleLibCompileToJarDebug FAILED
[        ] > Task :url_launcher:generateDebugRFile
[        ] > Task :package_info:generateDebugRFile
[ +100 ms] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[        ] Use '--warning-mode all' to show the individual deprecation warnings.
[        ] See https://docs.gradle.org/6.1.1/userguide/command_line_interface.html#sec:command_line_warnings
[        ] 226 actionable tasks: 225 executed, 1 up-to-date
[        ] FAILURE: Build failed with an exception.
[        ] * What went wrong:
[        ] Execution failed for task ':background_fetch:bundleLibCompileToJarDebug'.
[        ] > Couldn't read file content: '/builds/beeple-mobile-app/beeple-mobile/build/background_fetch/intermediates/javac/debug/classes/com/transistorsoft/flutter/backgroundfetch/BackgroundFetchModule$FetchStreamHandler.class'.
[        ] * Try:
[        ] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[        ] * Get more help at https://help.gradle.org
[        ] BUILD FAILED in 2m 49s
[ +539 ms] Running Gradle task 'assembleDebug'... (completed in 170.3s, longer than expected)
`
and 

`           FAILURE: Build failed with an exception.
           * Where:
           Build file '/opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_local_notifications-3.0.2/android/build.gradle' line: 22
           * What went wrong:
           A problem occurred evaluating root project 'flutter_local_notifications'.
           > Failed to apply plugin [id 'com.android.internal.version-check']
              > Minimum supported Gradle version is 5.6.4. Current version is 5.6.2. If using the gradle wrapper, try editing the distributionUrl in /opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_local_notifications-3.0.2/android/gradle/wrapper/gradle-wrapper.properties to gradle-5.6.4-all.zip
           * Try:
           Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
           * Get more help at https://help.gradle.org
           BUILD FAILED in 4s
[        ] Running Gradle task 'assembleAarRelease'... (completed in 5.0s)
[  +12 ms] 
           FAILURE: Build failed with an exception.
           * Where:
           Build file '/opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_local_notifications-3.0.2/android/build.gradle' line: 22
           * What went wrong:
           A problem occurred evaluating root project 'flutter_local_notifications'.
           > Failed to apply plugin [id 'com.android.internal.version-check']
              > Minimum supported Gradle version is 5.6.4. Current version is 5.6.2. If using the gradle wrapper, try editing the distributionUrl in /opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_local_notifications-3.0.2/android/gradle/wrapper/gradle-wrapper.properties to gradle-5.6.4-all.zip
           * Try:
           Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
           * Get more help at https://help.gradle.org
           BUILD FAILED in 4s
[   +6 ms] "flutter apk" took 332,083ms.
[   +8 ms] The plugin flutter_local_notifications could not be built due to the issue above.
[   +2 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      buildPluginsAsAar (package:flutter_tools/src/android/gradle.dart:848:7)
           <asynchronous suspension>
           #2      buildGradleApp (package:flutter_tools/src/android/gradle.dart:261:11)
           #3      buildGradleApp (package:flutter_tools/src/android/gradle.dart:444:19)
           #4      _rootRunUnary (dart:async/zone.dart:1198:47)
           #5      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #6      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #7      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #8      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #9      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #10     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
           #11     _rootRun (dart:async/zone.dart:1190:13)
           #12     _CustomZone.run (dart:async/zone.dart:1093:19)
           #13     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #14     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #15     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #17     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #18     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
[  +16 ms] ensureAnalyticsSent: 1ms
[   +4 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
```
</details>

Here is the full build log for extra info 
[build_log.txt](https://github.com/flutter/flutter/files/5780353/build_log.txt)

**Comment by pedromassangocode** [comment]
Hi @Sadoge 
Starting from Flutter `1.25.0-5.0.pre.108` the minimum supported Gradle is 6.5.

See https://github.com/flutter/flutter/issues/71968#issuecomment-741690823.

**Comment by Sadoge** [comment]
@pedromassangocode Thanks for the quick response, sadly even after changing it to gradle 6.5, it still gives me the exact same error. Also I am using 1.22 not 1.25

**Comment by pedromassangocode** [comment]
Hi @Sadoge 
The issue is caused by a third-party plugin called `flutter_local_notifications`. Please file an issue on its [repository](https://github.com/MaikuB/flutter_local_notifications)!

```console
           FAILURE: Build failed with an exception.
           * Where:
           Build file '/opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_local_notifications-3.0.2/android/build.gradle' line: 22
           * What went wrong:
           A problem occurred evaluating root project 'flutter_local_notifications'.
           > Failed to apply plugin [id 'com.android.internal.version-check']
              > Minimum supported Gradle version is 5.6.4. Current version is 5.6.2. If using the gradle wrapper, try editing the distributionUrl in /opt/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_local_notifications-3.0.2/android/gradle/wrapper/gradle-wrapper.properties to gradle-5.6.4-all.zip
           * Try:
           Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
           * Get more help at https://help.gradle.org
           BUILD FAILED in 4s
[   +6 ms] "flutter apk" took 332,083ms.
[   +8 ms] The plugin flutter_local_notifications could not be built due to the issue above.
```

If you disagree, please write in the comments,  
providing your `flutter doctor -v`, your `flutter run -v`,  your `pubspec.yaml`
a minimal reproducible code sample that does not use 3rd party plugins,  
and I will reopen it.
Thank you

**Comment by Sadoge** [comment]
Finally was able to fix the issue. It apparently had to do with a "Too many open files issue". The error was not displayed on the screen but after searching for many hours we finally found where this issue can happen. Just increasing the amount of allowed opened files fixed the issue. Still unsure why the error was referring to this package specifically, maybe something that can be looked into as I found that more people are having this issue (wrong error messages on CI or locally)
